source/ui/dataman/AMScanThumbnailGridViewItemDelegate.cpp

```diff
@@ -0,0 +1,92 @@
+#include "AMScanThumbnailGridViewItemDelegate.h"
+#include "util/AMDateTimeUtils.h"
+
+AMScanThumbnailGridViewItemDelegate::AMScanThumbnailGridViewItemDelegate(QObject *parent) :
+	QItemDelegate(parent)
+{
+}
+
+void AMScanThumbnailGridViewItemDelegate::paint(QPainter *painter,
+												const QStyleOptionViewItem &option,
+												const QModelIndex &index) const
+{
+
+	if(option.rect.isEmpty() || !index.isValid())
+		return;
+
+	painter->save();
+
+	if(!index.parent().isValid()) {
+		// Is a scan
+
+		switch (index.column()) {
+		case 0:
+			paintItemBackground(painter, option);
+			break;
+		case 1:
+			paintText(painter, option, index.data(Qt::DisplayRole).toString());
+			break;
+		case 3:
+			paintText(painter, option,
+					  AMDateTimeUtils::prettyDateTime(index.data(Qt::DisplayRole).toDateTime()));
+			break;
+		case 4:
+			paintText(painter, option, index.data(Qt::DisplayRole).toString());
+			break;
+		default:
+			break;
+		}
+
+	} else {
+		// Is a thumbnail
+		switch (index.column()) {
+		case 1:
+			paintThumbnail(painter, option, index.data(Qt::DecorationRole).value<QPixmap>());
+			break;
+		case 2:
+			paintText(painter, option, index.data(Qt::DisplayRole).toString());
+			break;
+		default:
+			break;
+		}
+	}
+
+	painter->restore();
+}
+
+void AMScanThumbnailGridViewItemDelegate::paintThumbnail(QPainter *painter,
+															  const QStyleOptionViewItem &option,
+															  const QPixmap &pixmap) const
+{
+	painter->drawPixmap(option.rect, pixmap);
```

Comment:
When I altered the size values in the geometry manager it scaled the pixmaps to fit the rectangles. This wasn't ideal, as scaling the thumbnails of 1D plots more than a very small amount produced gaps in the lines. I ended up specifying a size in the geometry manager that didn't scale them enough that this effect occurred.